xskillscore/core/contingency.py

```diff
@@ -0,0 +1,740 @@
+            & (len(forecast_category_edges) - 1 == 2)
+            else False
+        )
+        self.table = self._get_contingency_table(dim)
+
+    def _get_contingency_table(self, dim):
+        table = histogram(
+            self.observations,
+            self.forecasts,
+            bins=[self.observation_category_edges, self.forecast_category_edges],
+            bin_names=[OBSERVATIONS_NAME, FORECASTS_NAME],
+            dim=dim,
+            bin_dim_suffix='_bin',
+        )
+
+        # Add some coordinates to simplify interpretation/post-processing
+        table = table.assign_coords(
+            {
+                OBSERVATIONS_NAME
+                + '_bin': _get_category_bounds(self.observation_category_edges)
+            }
+        ).rename({OBSERVATIONS_NAME + '_bin': OBSERVATIONS_NAME + '_category_bounds'})
+        table = table.assign_coords(
+            {FORECASTS_NAME + '_bin': _get_category_bounds(self.forecast_category_edges)}
+        ).rename({FORECASTS_NAME + '_bin': FORECASTS_NAME + '_category_bounds'})
+        table = table.assign_coords(
+            {
+                OBSERVATIONS_NAME
+                + '_category': (
+                    OBSERVATIONS_NAME + '_category_bounds',
+                    range(1, len(self.observation_category_edges)),
+                ),
+                FORECASTS_NAME
+                + '_category': (
+                    FORECASTS_NAME + '_category_bounds',
+                    range(1, len(self.forecast_category_edges)),
+                ),
+            }
+        )
+        table = table.swap_dims(
+            {
+                OBSERVATIONS_NAME + '_category_bounds': OBSERVATIONS_NAME + '_category',
+                FORECASTS_NAME + '_category_bounds': FORECASTS_NAME + '_category',
+            }
+        )
+
+        return table
+
+    def _sum_categories(self, categories):
+        """Returns sums of specified categories in contingency table
+
+            Parameters
+            ----------
+            category : str, optional
+                Contingency table categories to sum. Options are 'total', 'observations' and 'forecasts'
+
+            Returns
+            -------
+            Sum of all counts in specified categories
+
+        """
+
+        if categories == 'total':
+            N = self.table.sum(
+                dim=(OBSERVATIONS_NAME + '_category', FORECASTS_NAME + '_category'),
+                skipna=True,
+            )
+        elif categories == 'observations':
+            N = self.table.sum(dim=FORECASTS_NAME + '_category', skipna=True).rename(
+                {OBSERVATIONS_NAME + '_category': 'category'}
+            )
+        elif categories == 'forecasts':
+            N = self.table.sum(dim=OBSERVATIONS_NAME + '_category', skipna=True).rename(
+                {FORECASTS_NAME + '_category': 'category'}
+            )
+        else:
+            raise ValueError(f'"{categories}" is not a recognised category')
```

Comment:
I would add to this `"Pick one of ['total', 'observations', 'forecasts']` just for quality-of-life here.